### ISSUE TITLE
CMake compile flags issue

### ISSUE THREAD

**Issue by slouken** [libsdl-org/SDL#7074]
I noticed this in the FreeBSD CI build:
```
  [51/516] /usr/bin/cc -DSDL3_EXPORTS -DSDL_BUILD_MAJOR_VERSION=3 -DSDL_BUILD_MICRO_VERSION=0 -DSDL_BUILD_MINOR_VERSION=0 -DUSING_GENERATED_CONFIG_H -I/Users/runner/work/SDL/SDL/build/include-config- -I/Users/runner/work/SDL/SDL/src -I/Users/runner/work/SDL/SDL/build/include -I/Users/runner/work/SDL/SDL/include -I/Users/runner/work/SDL/SDL/include/SDL3 -I/usr/local/include -I/usr/local/include/libdrm -I/usr/local/include/libepoll-shim -I/Users/runner/work/SDL/SDL/build/wayland-generated-protocols -I/usr/local/include/dbus-1.0 -I/usr/local/lib/dbus-1.0/include -I/usr/local/include/ibus-1.0 -I/usr/local/include/glib-2.0 -I/usr/local/lib/glib-2.0/include -fPIC -fvisibility=hidden -mmmx -msse -msse2 -msse3 -I/usr/local/include -I/usr/local/include/pipewire-0.3 -I/usr/local/include/spa-0.2 -D_REENTRANT -I/usr/local/include/libdrm "-DHAVE_USBHID_H -DUSBHID_NEW" "-D_REENTRANT -D_THREAD_SAFE" -idirafter /Users/runner/work/SDL/SDL/src/video/khronos -Wall -fno-strict-aliasing -Wdeclaration-after-statement -Wdocumentation -Wdocumentation-unknown-command -Wshadow -fcolor-diagnostics -MD -MT CMakeFiles/SDL3.dir/src/dynapi/SDL_dynapi.c.o -MF CMakeFiles/SDL3.dir/src/dynapi/SDL_dynapi.c.o.d -o CMakeFiles/SDL3.dir/src/dynapi/SDL_dynapi.c.o -c /Users/runner/work/SDL/SDL/src/dynapi/SDL_dynapi.c
  In file included from <built-in>:343:
  <command line>:8:9: warning: '_REENTRANT' macro redefined [-Wmacro-redefined]
  #define _REENTRANT -D_THREAD_SAFE 1
          ^
  <command line>:6:9: note: previous definition is here
  #define _REENTRANT 1
          ^
  1 warning generated.
```

Notice that PTHREAD_CFLAGS and USB_CFLAGS are added with quotes to the command line?
Is there a way to expand them when setting target_compile_options()?

**Comment by madebr** [comment]
The warning is fixed in a3ae26ddd0a34ee3f5f39311dbdf420b24d95939.
The FreeBSD compiler error is addressed in 0785f272fd9eb9c2cb366b074b09417aeea3f3c6

**Comment by slouken** [comment]
Thanks!

We should probably do an audit of flags that are added to compilation. I found a similar issue with Raspberry Pi as well.
